File: sliding-window/javascript/21921.js
const [[n, X], arr] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .split("\n")
  .map((input) => input.split(" ").map(Number));

const accArr = arr.reduce((answer, num, idx) => {
  answer.push(answer[idx - 1] ? answer[idx - 1] + num : num);
  return answer;
}, []);
const rangeAccArray = accArr.map((num, idx) => (idx >= X ? num - accArr[idx - X] : num));
const maxNum = Math.max(...rangeAccArray);
const cnt = rangeAccArray.filter((num) => num === maxNum).length;

maxNum > 0 ? console.log([maxNum, cnt].join("\n")) : console.log("SAD");

File: sliding-window/python/21921.py
N,X = map(int,input().split())
arr = list(map(int,input().split()))
answer = [0,0]
total = 0
for i in range(len(arr)):
  total+= arr[i]
  if i >= X:
    total -= arr[i-X]
  if answer[0] <= total:
    if answer[0] == total:
      answer[1] += 1
      continue
    answer[0] = total
    answer[1] = 1

if answer[0] == 0:
  print("SAD")
else:
  print(answer[0])
  print(answer[1])

